celery/concurrency/asynpool.py

```diff
@@ -26,12 +26,12 @@
 from time import sleep
 from weakref import WeakValueDictionary, ref
 
+import dill as _pickle
 from billiard import pool as _pool
 from billiard.compat import isblocking, setblocking
 from billiard.pool import ACK, NACK, RUN, TERMINATE, WorkersJoined
 from billiard.queues import _SimpleQueue
 from kombu.asynchronous import ERR, WRITE
-from kombu.serialization import pickle as _pickle
```

Comment:
Thinking out load - shouldn’t the `import dill as _pickle` be actually inside `kombu.serialization` so if it is used in other places, this will apply the change to `dill` globally?
This might also mean that the only PR required is, in fact, in Kombu and not in Celery at all.

Hmmm…. though it might have additional side effects.. 🤔

Reply:
This could very well be a preferred place to do this modification, regarding this PR. When it comes to the side effects, as `dill` is supposed to be a [drop-in replacement](https://github.com/uqfoundation/dill?tab=readme-ov-file#basic-usage) as you mentioned earlier, I assume this should not have that many side effects, if any? Also, if the tests pass, this should be okay?

What would have been great is having the option to pass the serialazer as a parameter somewhere for it to be applied globally, whether it is `celery` or `billiard` (or even `kombu`). This would however require far more refactoring I think.

Reply:
Is this resolved? Additionally, would you have any idea about the non passing test suite?